src/lib.rs

```diff
@@ -61,10 +69,81 @@ impl BinaryQF {
     pub fn discriminant(&self) -> BigInt {
         // for negative delta we compute 4ac - b^2
         let abs_delta = BigInt::from(4) * &self.a * &self.c - &self.b * &self.b;
-        assert!(abs_delta > BigInt::zero());
+//        assert!(abs_delta > BigInt::zero());
         -abs_delta
     }
 
+    pub fn compressing_form(&self)->BinaryQF_CompressedForm {
```

Comment:
I think some reference explaining the construction is needed.

Reply:
BinaryQF_CompressedForm is a representation of an element in class group.  BinaryQF_CompressedForm is equivalent to BinaryQF except the "b" element which is substituted with a another parameter "b_compressed"  with half the space.  "b_compressed" can be later extracted to the original "b" using the function decompressing_form(BinaryQF_CompressedForm)